test/test_sparse_csr.py

```diff
@@ -90,113 +91,129 @@ def test_sparse_csr_print(self):
-                                      values, dtype=torch.double)
-        dense = torch.tensor([[1, 2, 1], [3, 4, 0]], dtype=torch.double)
+                                      values, dtype=dtype, device=device)
+        dense = torch.tensor([[1, 2, 1], [3, 4, 0]], dtype=dtype, device=device)
         self.assertEqual(csr.to_dense(), dense)
 
-    def test_coo_to_csr_convert(self):
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_coo_to_csr_convert(self, device, dtype):
         size = (5, 5)
-        dense = torch.randn(size)
+        dense = torch.randn(size, dtype=dtype, device=device)
         sparse_coo = dense.to_sparse()
         sparse_csr = sparse_coo.to_sparse_csr()
 
         self.assertTrue(sparse_csr.is_sparse_csr)
         self.assertEqual(sparse_csr.to_dense(), dense)
 
-        vec = torch.randn((5, 1))
+        vec = torch.randn((5, 1), dtype=dtype, device=device)
         coo_product = sparse_coo.matmul(vec)
         csr_product = sparse_csr.matmul(vec)
 
         self.assertEqual(coo_product, csr_product)
 
-        vec = torch.randn((100, 1))
+        vec = torch.randn((100, 1), dtype=dtype, device=device)
         index = self.index_tensor([
             [1, 0, 35, 14, 39, 6, 71, 66, 40, 27],
             [92, 31, 62, 50, 22, 65, 89, 74, 56, 34],
         ])
-        values = self.value_tensor([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])
-        coo = torch.sparse_coo_tensor(index, values, torch.Size([100, 100]))
+        values = torch.tensor([1, 2, 3, 4, 5, 6, 7, 8, 9, 10], dtype=dtype, device=device)
+        coo = torch.sparse_coo_tensor(index, values, torch.Size([100, 100]), dtype=dtype, device=device)
         csr = coo.to_sparse_csr()
 
         self.assertEqual(coo.matmul(vec), csr.matmul(vec))
 
-    def test_mkl_matvec_warnings(self):
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_mkl_matvec_warnings(self, device, dtype):
         if torch.has_mkl:
             sp = torch.sparse_csr_tensor(torch.tensor([0, 2, 4]),
                                          torch.tensor([0, 1, 0, 1]),
-                                         torch.tensor([1, 2, 3, 4], dtype=torch.double))
-            vec = torch.randn((2, 1))
+                                         torch.tensor([1, 2, 3, 4], dtype=dtype, device=device))
+            vec = torch.randn((2, 1), dtype=dtype, device=device)
             with warnings.catch_warnings(record=True) as w:
                 sp.matmul(vec)
                 self.assertEqual(len(w), 2)
 
-    def test_dense_convert_error(self):
+    @dtypes(torch.double)
+    def test_dense_convert_error(self, device, dtype):
         size = (4, 2, 4)
-        dense = torch.randn(size)
+        dense = torch.randn(size, dtype=dtype, device=device)
 
         with self.assertRaisesRegex(RuntimeError, "Only 2D"):
             sparse = dense.to_sparse_csr()
 
-    def test_csr_matvec(self):
+    @onlyCPU
+    @dtypes(torch.double)
+    def test_csr_matvec(self, device, dtype):
```

Comment:
You probably want to test matvec for both int32 and int64 index types.

Reply:
ok